common/bolt11.c

```diff
@@ -616,7 +616,14 @@ struct bolt11 *bolt11_decode(const tal_t *ctx, const char *str,
 		 */
 		b11->msat = NULL;
 	} else {
-		u64 m10 = 10;
+		/* BOLT #11:
+		 *
+		 * 2. amount: optional number in that currency, followed by an
+		 * optional `multiplier` letter. The unit encoded here is the
+		 * 'social' convention of a payment unit -- in the case of
+		 * Bitcoin the unit is 'bitcoin' NOT satoshis.
+		 */
+		u64 m10 = 10E12; /* Pico satoshis in a Bitcoin */
```

Comment:
It seems like you've got an off by 10x issue here. 
```
>       assert(b11['msatoshi'] == 7*10**11)
E       assert 7000000000000 == 700000000000
E         -7000000000000
E         +700000000000
```

Reply:
Yes, this should be `10 * MSAT_PER_BTC` (which follows the format of the `multipliers` table).